File: python/832 Flipping an Image.py
# Runtime: 76ms

class Solution(object):
	def flipAndInvertImage(self, A):
		"""
		:type A: List[List[int]]
		:rtype: List[List[int]]
		"""
		if len(A) == 0:
			return A
		res = []
		for row in A:
			res.append([])  # 每行先加一个空list
			for col in row[::-1]:
				tmp = 1 - col # 实现很简单
				res[-1].append(tmp)
		return res

File: java/832_flipping_an_image.java
// Runtime: 8 mss

class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        int i = 0;
        for(int[] row : A){
            int[] f = new int[row.length];
            for(int m = 0;m<row.length;m++){
                f[m] = row[row.length -1 -m];
            }

            for(int j=0; j<f.length; j++){
                f[j] = 1 - f[j];
            }
            A[i] = f;
            i++;
        }
        return A;
    }
}